recipes/cpython/all/conanfile.py

```diff
@@ -53,286 +56,294 @@ class CPythonConan(ConanFile):
+                raise ConanInvalidConfiguration("Static msvc build disabled (>=3.10) due to \"AttributeError: module 'sys' has no attribute 'winver'\"")
+
+        if self.options.get_safe("with_curses", False) and not self.dependencies["ncurses"].options.with_widec:
+            raise ConanInvalidConfiguration("cpython requires ncurses with wide character support")
+
+        if self._supports_modules:
+            if Version(self.version) >= "3.9.0":
+                if self.dependencies["mpdecimal"].ref.version < Version("2.5.0"):
+                    raise ConanInvalidConfiguration("cpython 3.9.0 (and newer) requires (at least) mpdecimal 2.5.0")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def _generate_autotools(self):
+        tc = AutotoolsToolchain(self, prefix=self.package_folder)
+        # Not necessary, just cleans up the output
+        tc.update_configure_args({"--enable-static": None, "--disable-static": None})
         yes_no = lambda v: "yes" if v else "no"
-        conf_args = [
-            "--enable-shared={}".format(yes_no(self.options.shared)),
+        tc.configure_args += [
             "--with-doc-strings={}".format(yes_no(self.options.docstrings)),
             "--with-pymalloc={}".format(yes_no(self.options.pymalloc)),
             "--with-system-expat",
             "--with-system-ffi",
             "--enable-optimizations={}".format(yes_no(self.options.optimizations)),
             "--with-lto={}".format(yes_no(self.options.lto)),
             "--with-pydebug={}".format(yes_no(self.settings.build_type == "Debug")),
+            "--with-system-libmpdec",
+            "--with-openssl={}".format(self.dependencies["openssl"].package_folder),
         ]
-        if self._is_py2:
-            conf_args.extend([
-                "--enable-unicode={}".format(yes_no(self.options.unicode)),
-            ])
-        if self._is_py3:
-            conf_args.extend([
-                "--with-system-libmpdec",
-                "--with-openssl={}".format(self.deps_cpp_info["openssl"].rootpath),
-                "--enable-loadable-sqlite-extensions={}".format(yes_no(not self.options["sqlite3"].omit_load_extension)),
-            ])
-        if self.settings.compiler == "intel":
-            conf_args.extend(["--with-icc"])
-        if tools.get_env("CC") or self.settings.compiler != "gcc":
-            conf_args.append("--without-gcc")
+        if Version(self.version) >= "3.10":
+            tc.configure_args.append("--disable-test-modules")
+        if self.options.get_safe("with_sqlite3"):
+            tc.configure_args.append("--enable-loadable-sqlite-extensions={}".format(
+                yes_no(not self.dependencies["sqlite3"].options.omit_load_extension)
+            ))
+        if self.settings.compiler == "intel-cc":
+            tc.configure_args.append("--with-icc")
+        if os.environ.get("CC") or self.settings.compiler != "gcc":
```

Comment:
```suggestion
        if self.settings.compiler != "gcc":
```

Avoid using external environment variables, it will result in not reproducible scenarios.

Reply:
This was here prior, I'll double check that it doesn't break anything if used.

Reply:
I think I can actually remove this condition entirely, seems that the flag was removed in 3.8: https://github.com/python/cpython/blob/main/Misc%2FNEWS.d%2F3.8.0a1.rst#L7485

Reply:
Much better, please, remove then. Thank you for checking it!